src/libstore/build/local-derivation-goal.cc

```diff
@@ -2327,10 +2330,6 @@ DrvOutputs LocalDerivationGoal::registerOutputs()
             }, *orifu);
         }},
         {[&](const std::string & path, const std::string & parent) {
-            // as the result is cast to an Error, the handler for BuildErrors is not run
-            // this will move paths out of a chroot directory
-            cleanupError();
```

Comment:
This was removed line was added to the wrong commit. I think it should be part of the previous one, or it should be an extra commit, although it seems that that would create a potentially broken revision for `git bisect`.

Reply:
You're right, it should have been added to `Throw error inside closure`. I have done a rebase to add it to the correct commit (first time doing an interactive rebase)